### PR TITLE
new class to summarize basket

### PR DESCRIPTION
**Pull request by niewicz**
new version of basket with sinatra
@gotar

### CODE REVIEW

File: shop.rb [hunk 1]
```diff
@@ -31,8 +31,9 @@ class App < Sinatra::Base
     end
 
     get "/basket" do
-      basket = FetchBasket.new.call
-      erb :"basket/show", locals: {basket: basket}
+      basket_summary = FetchBasketSummary.new.call
+      p basket_summary
```
Comment (niewicz) [inline shop.rb]: printowanie basket_summary już usunięte

File: Gemfile [hunk 2]
```diff
@@ -3,5 +3,6 @@ source "https://rubygems.org"
 gem "sinatra"
 
 group :test do
+  gem "rack-test"
 	gem "rspec"
```
Comment (gotar) [inline Gemfile]: wrong indentation

File: lib/model/basket_summary.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+require_relative "../services/fetch_basket"
+require_relative "../services/fetch_product"
+
+module Shop
+
```
Comment (gotar) [inline lib/model/basket_summary.rb]: drop empty line

File: lib/model/basket_summary.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      calculate_totals
+      @sum_net = calc_sum_net
+      @sum_gross = calc_sum_gross
+    end
+
+    private
+    def create_products_list
+      basket = FetchBasket.new.call
+      basket.each do |basket_item|
+        @products_list << {product: FetchProduct.new.call(basket_item.product_id), quantity: basket_item.quantity, total_net: 0, total_gross: 0}
```
Comment (gotar) [inline lib/model/basket_summary.rb]: too long line, limit to max 80 chars

File: lib/services/fetch_basket_summary.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+require_relative "../model/basket_summary"
+
+module Shop
+
```
Comment (gotar) [inline lib/services/fetch_basket_summary.rb]: drop empty line

File: lib/services/fetch_basket_summary.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+  class FetchBasketSummary
+    def call
+      BasketSummary.new
+    end
+  end
+
```
Comment (gotar) [inline lib/services/fetch_basket_summary.rb]: drop empty line

File: lib/services/fetch_basket_summary.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+end
```
Comment (gotar) [inline lib/services/fetch_basket_summary.rb]: add new line

**Pull request merged**